unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/BodyShapeController.cs

```diff
@@ -18,6 +18,12 @@ public class BodyShapeController : WearableController, IBodyShapeController
     private Transform animationTarget;
     private bool facialFeaturesVisible;
     private HashSet<string> currentHiddenList = new HashSet<string>();
+    public Transform[] bones;
+    public Transform rootBone;
+
+    private bool lowerBodyActive = true;
+    private bool upperBodyActive = true;
+    private bool feetActive = true;
```

Comment:
`SetActiveParts` was conflicting issues with `UpdateVisibility` and could led to invalid avatars if they were called in the wrong order. Made `SetActiveParts` just write to these variables, so the visibility is taken into account everywhere and no issues could be arising from bad calling order anymore.